src/nodes.rs

```diff
@@ -86,6 +86,110 @@ pub async fn create_nodes(
     nodes
 }
 
+pub async fn create_nodes_filtered(
+    network_type_filter: NetworkType,
+    indices: &NodesIndices,
+    node_addrs: &[SocketAddr],
+    node_network_types: &[NetworkType],
+    geo_cache: &GeoIPCache,
+) -> Vec<Node> {
+    let num_nodes = indices.len();
+    let mut index_map: Vec<i32> = vec![0; num_nodes];
+
+    // Create reindexing map using filter value
+    //    a) the nodes we keep get new indexing, 0..N
+    //    b) the nodes we don't want get -1
+    let mut index: i32 = 0;
+    for (n, network_type) in node_network_types.iter().enumerate() {
+        if network_type_filter == *network_type {
+            index_map[n] = index;
+            index += 1;
+        } else {
+            index_map[n] = -1;
```

Comment:
we can avoid this `else` block if we initialize all indices in the map to -1 from the start:
```
let mut index_map: Vec<i32> = vec![-1; num_nodes];
```

so the starting assumption is: we don't need the index_map index until we need it - but then we set it on the line 105

Reply:
ah, good idea... I missed that.

Reply:
done